go-controller/pkg/ovn/libovsdbops/meter.go

```diff
@@ -0,0 +1,71 @@
+package libovsdbops
+
+import (
+	"context"
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+// FindMeterByName finds an ovn meter by name, will return `errNotFound` if it does not exist
+func FindMeterByName(nbClient libovsdbclient.Client, name string) (*nbdb.Meter, error) {
+	searched := &nbdb.Meter{
+		Name: name,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+	err := nbClient.Get(ctx, searched)
+	if err != nil {
+		return nil, err
+	}
+
+	return searched, nil
+}
+
+// UpdateMeterFairness updates the `fair` column of an ovn Meter, should only be called if a
+// full in memory version of the meter has already been acquired with FindMeter
+func UpdateMeterFairness(nbClient libovsdbclient.Client, meter *nbdb.Meter, fairness bool) error {
+	meter.Fair = &fairness
+
+	opModel := OperationModel{
+		Model: meter,
+		OnModelUpdates: []interface{}{
+			&meter.Fair,
+		},
+		ErrNotFound: true,
+	}
+
+	m := NewModelClient(nbClient)
+	if _, err := m.CreateOrUpdate(opModel); err != nil {
+		return fmt.Errorf("error while updating Meter Fairness to: %v error %v", fairness, err)
+	}
+
+	return nil
+}
+
+// CreateMeterWithBand simulates the ovn-nbctl operation performed by the `meter-add` command. i.e create the Meter
+// and accompanying meter_band, and add the meterband to the meter only call this if the meter does not exist
+func CreateMeterWithBand(nbClient libovsdbclient.Client, meter *nbdb.Meter, meterBand *nbdb.MeterBand) error {
+	opModels := []OperationModel{
+		{
+			Model: meterBand,
+			DoAfter: func() {
+				meter.Bands = []string{meterBand.UUID}
+			},
+		},
+		{
+			Model: meter,
```

Comment:
in the unlikely scenario that meter already had a meterBand, consider making this a mutateInsert opeartion?
Otherwise maybe we should call this fcn as ... 'CreateMeterAndSetMeterBand' ?!? Maybe not, that is ugly :P

Reply:
Well we assume if we're in the function the meter didn't exist, and we're also assuming that if the meter didn't exist that the meterBand doesn't exist either (since there's no way to lookup the meterband without looking up the meter)